File: backend/app/schemas/application_packet.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ApplicationPacketRequest(BaseModel):
    model_config = ConfigDict(extra="forbid")

    update_decision: bool = True
    include_resume_bullets: bool = True
    include_cover_note_outline: bool = True
    include_cold_dm_outline: bool = True
    include_checklist: bool = True
    include_risk_review: bool = True


class ApplicationPacketItem(BaseModel):
    label: str
    value: str
    reason: str


class ApplicationPacketSection(BaseModel):
    title: str
    items: list[ApplicationPacketItem]


class ApplicationPacketResponse(BaseModel):
    job_id: str
    decision_id: str | None = None
    company_name: str | None = None
    title: str
    role_category: str | None = None
    match_tier: str | None = None
    total_score: float | None = None
    remote_eligibility: str | None = None
    application_positioning: str
    resume_focus: list[ApplicationPacketItem]
    resume_bullet_suggestions: list[ApplicationPacketItem]
    project_evidence_to_use: list[ApplicationPacketItem]
    cover_note_outline: ApplicationPacketSection | None = None
    cold_dm_outline: ApplicationPacketSection | None = None
    application_checklist: list[ApplicationPacketItem]
    risks_to_verify: list[ApplicationPacketItem]
    suggested_apply_plan: list[ApplicationPacketItem]
    generated_at: datetime

File: frontend/types/application-packet.ts
export type ApplicationPacketRequest = {
  update_decision?: boolean;
  include_resume_bullets?: boolean;
  include_cover_note_outline?: boolean;
  include_cold_dm_outline?: boolean;
  include_checklist?: boolean;
  include_risk_review?: boolean;
};

export type ApplicationPacketItem = {
  label?: string | null;
  value?: string | null;
  reason?: string | null;
  severity?: string | null;
};

export type ApplicationPacketEvidenceItem = ApplicationPacketItem | string | Record<string, unknown>;

export type ApplicationPacketSection = {
  title?: string | null;
  items?: ApplicationPacketItem[] | null;
};

export type ApplicationPacketResponse = {
  job_id: string;
  decision_id?: string | null;
  company_name?: string | null;
  title?: string | null;
  role_category?: string | null;
  match_tier?: string | null;
  total_score?: number | null;
  remote_eligibility?: string | null;
  resume_id?: string | null;
  resume_used?: boolean;
  resume_match_summary?: string | null;
  resume_strengths?: ApplicationPacketEvidenceItem[] | null;
  resume_gaps?: ApplicationPacketEvidenceItem[] | null;
  resume_bullet_sources?: ApplicationPacketEvidenceItem[] | null;
  application_positioning?: string | null;
  resume_focus?: ApplicationPacketItem[] | null;
  resume_bullet_suggestions?: ApplicationPacketItem[] | null;
  project_evidence_to_use?: ApplicationPacketItem[] | null;
  cover_note_outline?: ApplicationPacketSection | null;
  cold_dm_outline?: ApplicationPacketSection | null;
  application_checklist?: ApplicationPacketItem[] | null;
  risks_to_verify?: ApplicationPacketItem[] | null;
  suggested_apply_plan?: ApplicationPacketItem[] | null;
  generated_at?: string | null;
};
